controllers/v1/criteria.js

```diff
@@ -0,0 +1,1319 @@
+                        }
+                        else {
+                            bodyParam.filter.value = submissionId;
+                        }
+                    }
+                    else {
+                        bodyParam.filter.value = submissionId;
+                    }
+
+                    //pass the query as body param and get the resul from druid
+                    let options = config.druid.options;
+                    options.method = "POST";
+                    options.body = bodyParam;
+                    let data = await rp(options);
+
+                    if (!data.length) {
+                        resolve({
+                            "result": false,
+                            "data": "SUBMISSION_ID_NOT_FOUND"
+                        });
+                    } else {
+
+                        let reportType = "criteria";
+                        let chartData = await helperFunc.instanceReportChart(data, reportType);
+
+                        //Get evidence data from evidence datasource
+                        let inputObj = {
+                            submissionId: submissionId
+                        }
+
+                        let evidenceData = await getEvidenceData(inputObj);
+                        let responseObj;
+
+                        if (evidenceData.result) {
+                            responseObj = await helperFunc.evidenceChartObjectCreation(chartData, evidenceData.data, req.headers["x-auth-token"]);
+                        } else {
+                            responseObj = chartData;
+                        }
+
+                        responseObj = await helperFunc.getCriteriawiseReport(responseObj);
+
+                        resolve(responseObj);
+
+                    }
+                })
+                .catch(function (err) {
+                    let response = {
+                        result: false,
+                        message: 'INTERNAL_SERVER_ERROR'
+                    };
+                    resolve(response);
+                });
+        }
+    });
+};
+
+
+
+//Funcion for instance observation pdf generation
+async function instancePdfReport(req, res) {
+  
+  return new Promise(async function (resolve, reject) {
+
+    let instaRes = await instanceReportData(req, res);
+  
+    if (("observationName" in instaRes) == true) {
+
+      let resData = await pdfHandler.instanceCriteriaReportPdfGeneration(instaRes, true);
+
+      let hostname = req.headers.host;
+
+      resData.pdfUrl = "https://" + hostname + "/dhiti/api/v1/criteria/pdfReportsUrl?id=" + resData.pdfUrl
+
+      resolve(resData);
+    }
+
+    else {
+      resolve(instaRes);
+    }
+
+  });
+};
+
+
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceScoreReport 
+   * Instance score report
+   * @apiVersion 1.0.0
+   * @apiGroup Criteria
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "submissionId": "",
```

Comment:
@deepakr123 - Add filter in request body

Reply:
done